src/cursor/cursor.ts

```diff
@@ -54,6 +54,81 @@ export interface CursorOptions extends CoreCursorOptions {
   numberOfRetries?: number;
 }
 
+const kCursor = Symbol('cursor');
+
+export class CursorStream extends Readable {
+  [kCursor]: Cursor;
+
+  /** @event */
+  static readonly CLOSE = 'close' as const;
+  /** @event */
+  static readonly DATA = 'data' as const;
+  /** @event */
+  static readonly END = 'end' as const;
+  /** @event */
+  static readonly FINISH = 'finish' as const;
+  /** @event */
+  static readonly ERROR = 'error' as const;
+  /** @event */
+  static readonly PAUSE = 'pause' as const;
+  /** @event */
+  static readonly READABLE = 'readable' as const;
+  /** @event */
+  static readonly RESUME = 'resume' as const;
```

Comment:
I think most of these can go away, right? Aren't they events that we inherit from `Readable`?

Reply:
I suppose we could take them away, `finish` is the only non-standard one. However, it's nice getting all the supported events as constants in code completion when I start typing `myCursorStream.on(CursorStream.`, which goes away if we remove these.